src/storage/store/node_table.cpp

```diff
@@ -392,7 +392,7 @@ void NodeTable::serialize(Serializer& serializer) const {
 
 bool NodeTable::isVisible(const Transaction* transaction, offset_t offset) const {
     auto [nodeGroupIdx, offsetInGroup] = StorageUtils::getNodeGroupIdxAndOffsetInChunk(offset);
-    auto* nodeGroup = getNodeGroupNoLock(nodeGroupIdx);
+    auto* nodeGroup = getNodeGroup(nodeGroupIdx);
```

Comment:
Ran into occasional segfaults due to nodeGroup because sometimes NULL (when running with many threads). Let me know if anyone has any more insight on this but I think there might have previously been some data races here (maybe on reallocating the nodeGroup vector, etc.)

Reply:
Nice catch! You're right, there can be race conditions during node batch insert.
However, the change here can cause performance regression with index lookups, especially during copy rel pipeline, we need to do lookups to figure out bound/dst node offsets first, under which case, locking is not required.

Can u add `isVisibleNoLock` and let `lookupPK` go through that?

Reply:
updated